### DISCUSSION TITLE
Increasing text contrast

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#5616] [Discussion]
Currently, the Scratch site has a few things that could be better in terms of text contrast. There are some UI elements that can be hard for some people to read. This discussion is meant to be a place to talk about resolving that issue for both Scratch and Scratch Addons with Scratch Addons features.

## Goals
- [ ] Offer ways to improve contrast on the Scratch website
- [ ] Improve the contrast in Scratch Addons

#### Offering a "high contrast text" option (#5895)

In the customizable colors addons, this could be a setting to lower the black text "cut-off" (in that you wouldn't have to make background colors as bright for the text to become black). This would probably result in the default navigation bar and messages badge color making the text turn black. In addition, black text would be made darker and white text would be made brighter.

Notable examples of where this could help:
- The navigation bar
- The messages badge
- The Share button
- The Cancel button below the new comment field
- All blue buttons

#### Offering high contrast presets (#5539)

Make a high contrast preset in the customizable colors addons.

#### Allow customizing more elements

This would result in the user being able to change more elements to colors that give the text higher contrast.

#### Adjust UI elements for better contrast

We could make a few non-optional changes to make the contrast better for everyone in ways that most people wouldn't care to accept, and/or offer an option for more extreme contrast changes.

Notable examples of where this could help:
- Text on orange buttons and backgrounds
- Category badges (discussed in https://github.com/ScratchAddons/ScratchAddons/issues/5462)

---

Here's a bonus link to a contrast checker I use: https://contrastchecker.com/

You can find accessibility-related issues we're working on [here](https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20accessibility).

**Comment by Secret-chest** [comment]
BTW, you should be using [the APCA contrast checker](https://www.myndex.com/APCA/?BG=ff7b26&TXT=ffffff&DEV=G4g&BUF=A22) since it is a much more accurate representation of human vision and also seeks factors such as text size and weight.
